core/src/main/java/de/jplag/Submission.java

```diff
@@ -247,6 +247,11 @@ private static File createErrorDirectory(String... subdirectoryNames) {
 
         try {
             tokenList = language.parse(new HashSet<>(files));
+            for (Token token : tokenList) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug(String.join(" | ", token.getType().toString(), Integer.toString(token.getLine()), token.getSemantics().toString()));
+                }
+            }
```

Comment:
We could probably move the loop into the if condition in order to avoid iterating whenever debug is not enabled.

Reply:
Good catch 👍